src/app/games/connect-six/tests/ConnectSixMinimax.spec.ts

```diff
@@ -0,0 +1,169 @@
+
+    beforeEach(() => {
+        const rules: ConnectSixRules = ConnectSixRules.get();
+        minimax = new ConnectSixMinimax(rules, 'Minimax');
+    });
+    it('should propose only one move at first turns', () => {
+        // Given the initial node
+        const width: number = ConnectSixState.WIDTH;
+        const height: number = ConnectSixState.HEIGHT;
+        const state: ConnectSixState = ConnectSixState.getInitialState();
+        const node: ConnectSixNode = new ConnectSixNode(state);
+
+        // When asking it the list of move at the first turn
+        const moves: ConnectSixMove[] = minimax.getListMoves(node);
+
+        // Then it should only include the center of the board
+        const cx: number = Math.floor(width/2);
+        const cy: number = Math.floor(height/2);
+        expect(moves.length).toBe(1);
+        expect(moves[0]).toEqual(ConnectSixFirstMove.from(new Coord(cx, cy)));
+    });
+    it('should do winning move when one is possible', () => {
+        // Given a board where there is place for a victory of first player
+        const board: Table<PlayerOrNone> = [
+            [O, O, O, O, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+        ];
+        const state: ConnectSixState = new ConnectSixState(board, 2);
+        const node: ConnectSixNode = new ConnectSixNode(state);
+
+        // When asking what is the best move
+        const bestMove: ConnectSixMove = node.findBestMove(1, minimax);
+
+        // Then it should be that victory
+        expect(bestMove).toEqual(ConnectSixDrops.from(new Coord(4, 0), new Coord(5, 0)).get());
+    });
+    it('should block double-open fives at level two', () => {
```

Comment:
le terme technique c'est open five vs. closed five, non ?

Reply:
Je suis pas certains, quelle littérature faire autoritée ?

Reply:
la page wikipédia de Gomoku : https://en.wikipedia.org/wiki/Gomoku
Et me semble avoir lu avant la description du créateur de connect 6 qui parlait justement d'open five / closed five, mais je ne retombe point dessus

Reply:
Oui mais ici on parle d'un jeu ou tu drop deux pièces par tour, contrairement à Pente et Gomoku, d'où la distinction ...
Il existerais alors, propose-je:
A.  double open: _ _ X X X X _ _
B. open: O _ X X X X _ O
C. cross-open: O _ X X X X _ _ 
D. closed: O X X X X _ O avec ou sans _ d'ailleurs
Mais on pourrait  débattre que l'importance c'est de distinguer par nombre de blocage nécessaire: 2 pour A, 1 pour B&C, 0 pour D.

Reply:
alors faut décrire ces conventions quelque part ! Car clairement j'aurais pas su que "double open" faisait pas référence à B, d'ailleurs le nom est pas idéal !

Mais ça dévie trop du vocabulaire original: closed à Gomoku c'est `OXXX`, ici soudainement c'est un "cross open", et closed signifie quelque chose de complètement différent.

Après, faut se poser la question de l'utilité d'introduire du vocabulaire pour juste décrire un seul test. Peut-être simplement renommer le test en "should block a specific configuration where [...]" avec une description plus claire (ou pas de description dans le it, et une explication en commentaire)

Reply:
Bah, un coup par tour ou deux coups, ça change à peu près TOUT les cas possibles.
Par open est ce qu'au Gomoku ça veut bien dire "il peut gagner si il joue là ?" ?
A. trois choix de comment bloquer en deux mouvements
B. deux choix de comment bloquer un un mouvement
C. un seul choix de blocage (mais si t'es con tu peux faire autre chose mais en deux mouvements)
D. ta menace est déjà morte

Je suis beaucoup plus chaud de mettre ça (enfin celles du message précédent) quelque part, dit moi juste où.

Reply:
met le ici alors, car c'est seulement utile pour décrire ce test ? (d'où la peut-être inutilité d'introduire ce vocabulaire pour un seul test)